src/aarch64/ffi.c

```diff
@@ -927,10 +965,37 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *orig_rvalue,
 #endif
     }
 
+#ifdef __CHERI_PURE_CAPABILITY__
+  /*
+   * Narrow bounds of c9 to on-stack arguments excluding argument structure
+   * objects passed by value in the function call which were replaced by
+   * pointers to the objects in the on-stack arguments.
+   *
+   * We can use precise bounds for the stack capability because it is already
+   * aligned to a larger size than its effective bounds.
+   */
+  if (state.nsaa > 0) {
+    FFI_ASSERT(SUBOBJECT_LENGTH(state.nsaa) <= stack_length);
+    FFI_ASSERT(stack == __align_up(stack, SUBOBJECT_ALIGNMENT(state.nsaa)));
+    context->x9 = (XREG)__builtin_cheri_bounds_set_exact(stack,
+      SUBOBJECT_LENGTH(state.nsaa));
+  } else if (isvariadic) {
```

Comment:
Might as well always initialise it rather than leave it as 0xaa.... from the padding, given we unconditionally use it in ffi_call_SYSV?

Reply:
Sounds good.